react/components/SKUSelector/index.js

```diff
@@ -1,110 +1,150 @@
-          secondaryVariation.name,
-          filteredSkus
-        )
-      }
-
-      return { mainVariation, secondaryVariation }
-    },
-    [parsedItems, skuSelected]
-  )
-
-  useEffect(() => {
-    const sku = (skuSelected && parseSku(skuSelected)) || parsedItems[0]
-
-    const { mainVariation, secondaryVariation } = buildVariations(sku)
-
-    setMainVariation(mainVariation)
-    setSecondaryVariation(secondaryVariation)
-  }, [skuSelected, parsedItems, buildVariations])
-
-  const redirectToSku = (skuId, isMainVariation) => {
+  const { setQuery } = useRuntime()
+  const redirectToSku = (skuId) => {
     setQuery(
       { skuId },
       {
-        replace: !isMainVariation,
+        replace: true,
       }
     )
   }
+  const [selectedSkuId, setSelectedSkuId] = useState(null)
+  const [selectedVariations, setSelectedVariations] = useState(null)
+  useEffect(() => {
+    const initalVariations = skuSelected ? selectedVariationFromItem(parseSku(skuSelected), variations) : buildEmptySelectedVariation(variations)
+    skuSelected && setSelectedSkuId(skuSelected.itemId)
+    setSelectedVariations(initalVariations)
+  }, [])
+  
+  const imagesMap = useImagesMap(parsedItems, variations)
+  
+  const onSelectItem = useCallback(({ name: variationName, value: variationValue, skuId, isMainAndImpossible, possibleItems }) => {
+    const isRemoving = selectedVariations[variationName] === variationValue
+    const newSelectedVariation = !isMainAndImpossible ? 
+      {
+      ...selectedVariations,
+      [variationName]: isRemoving ? null : variationValue,
+      } : 
+      { 
+        ...buildEmptySelectedVariation(variations),
+        [variationName]: variationValue 
+      }
+    // Set here for a better response to user
+    setSelectedVariations(newSelectedVariation)
+    const uniqueOptions = isRemoving ? {} : uniqueOptionToSelect(possibleItems, newSelectedVariation, isMainAndImpossible)
+    const finalSelected = {
+      ...newSelectedVariation,
+      ...uniqueOptions,
+    }
+    if (!isEmpty(uniqueOptions)) {
+      setSelectedVariations(finalSelected)
+    }
+    
+    const selectedNotNull = filter(Boolean, finalSelected)
+    const selectedCount = Object.keys(selectedNotNull).length
+    const variationsCount = Object.keys(variations).length
+    const allSelected = selectedCount === variationsCount
+    let skuIdToRedirect = skuId
+    if (!skuIdToRedirect || !isEmpty(uniqueOptions)) {
+      const newItem = findItemWithSelectedVariations(possibleItems, finalSelected)
+      skuIdToRedirect = newItem.itemId
+    }
 
-  const handleSkuSelection = useCallback((isMainVariation, skuId) => {
-    const sku = parsedItems.find(({ itemId }) => itemId === skuId)
-    const {
-      secondaryVariation: { options },
-    } = buildVariations(sku)
-
-    const isSecondaryPicked =
-      !isMainVariation || (options && options.length === 1)
+    if (selectedSkuId === skuIdToRedirect || isRemoving) {
+      // do nothing
```

Comment:
⛏ comment on why it does nothing, rather than what is being done (which is clear enough)